ztools/ztls/handshake_extensions.go

```diff
@@ -0,0 +1,378 @@
+package ztls
+
+import (
+	"errors"
+	"fmt"
+)
+
+type NullExtension struct {
```

Comment:
This file should all be pointer receivers, unless you have a good reason not to.